Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractRandomCutTree.java

```diff
@@ -146,6 +146,18 @@ static Cut randomCut(Random random, AbstractBoundingBox<?> box) {
         throw new IllegalStateException("The break point did not lie inside the expected range");
     }
 
+    /**
+     * A local version of randomcut that can be overridden for lower precision
+     * requirments
+     * 
+     * @param random random number generator
+     * @param box    bounding box corresponding to the generation of the cut
+     * @return a cut corresponding to the box
+     */
+    Cut treeCut(Random random, AbstractBoundingBox<?> box) {
```

Comment:
We already have a method named `randomCut` that can be overridden. Why do we need a new method with a new name here? I think the name `treeCut` is misleading if this does the same thing as `randomCut`. It sounds like this would prune branches off the tree.

This should be `protected` if we want it to be overridden by subclasses.

Reply:
RandomCut a static function. We can make that protected ...